docs/source/reference/kubernetes/kubernetes-deployment.rst

```diff
@@ -0,0 +1,235 @@
+.. _kubernetes-deployment:
+
+Deployment Guides
+-----------------
+Below we include minimal guides to set up a new Kubernetes cluster in different environments, including hosted services on the cloud.
+
+.. grid:: 2
+    :gutter: 3
+
+    .. grid-item-card::  Local Development Cluster
+        :link: kubernetes-setup-kind
+        :link-type: ref
+        :text-align: center
+
+        Run a local Kubernetes cluster on your laptop with ``sky local up``.
+
+    .. grid-item-card::  On-prem Clusters (RKE2, K3s, etc.)
+        :link: kubernetes-setup-onprem
+        :link-type: ref
+        :text-align: center
+
+        For on-prem deployments with kubeadm, RKE2, K3s or other distributions.
+
+    .. grid-item-card:: Google Cloud - GKE
+        :link: kubernetes-setup-gke
+        :link-type: ref
+        :text-align: center
+
+        Google's hosted Kubernetes service.
+
+    .. grid-item-card::  Amazon - EKS
+        :link: kubernetes-setup-eks
+        :link-type: ref
+        :text-align: center
+
+        Amazon's hosted Kubernetes service.
+
+.. _kubernetes-setup-kind:
+
+
+Deploying locally on your laptop
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To try out SkyPilot on Kubernetes on your laptop or run SkyPilot
+tasks locally without requiring any cloud access, we provide the
+:code:`sky local up` CLI to create a 1-node Kubernetes cluster locally.
+
+Under the hood, :code:`sky local up` uses `kind <https://kind.sigs.k8s.io/>`_,
+a tool for creating a Kubernetes cluster on your local machine.
+It runs a Kubernetes cluster inside a container, so no setup is required.
+
+1. Install `Docker <https://docs.docker.com/engine/install/>`_ and `kind <https://kind.sigs.k8s.io/>`_.
+2. Run :code:`sky local up` to launch a Kubernetes cluster and automatically configure your kubeconfig file:
+
+   .. code-block:: console
+
+     $ sky local up
+
+3. Run :code:`sky check` and verify that Kubernetes is enabled in SkyPilot. You can now run SkyPilot tasks on this locally hosted Kubernetes cluster using :code:`sky launch`.
+4. After you are done using the cluster, you can remove it with :code:`sky local down`. This will terminate the KinD container and switch your kubeconfig back to it's original context:
+
+   .. code-block:: console
+
+     $ sky local down
+
+.. note::
+    We recommend allocating at least 4 or more CPUs to your docker runtime to
```

Comment:
Allocating `4 or more CPUs to your docker runtime` is a bit vague, as we did not mention allocating CPUs in the previous steps, and the link to the docker instruction does not directly point to the instruction for how to allocate more CPUs.

How about we simply saying that more than 4 CPU cores are required for the local machine to have the local cluster functional?

Reply:
Hmm, even if the user has 4+ CPUs on their machine, their docker engine may be configured to use fewer CPUs. I have updated the link to kind's docs where they have instructions with screenshots. Does that help?